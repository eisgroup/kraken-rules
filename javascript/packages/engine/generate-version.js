const fs = require('fs')
const path = require('path')
const xmlParser = require('fast-xml-parser')

console.log('----------------------------------');
console.log('generating engine version constant');
console.log('----------------------------------');


const pomPath = path.join(
    __dirname,
    '../',
    '../',
    '../',
    'pom.xml'
)

console.log(`  reading '${pomPath}'`);

const pom = xmlParser.parse(
    fs.readFileSync(
        pomPath,
        { encoding: 'utf8' }
    )
)

console.log(`  version is: '${pom.project.version}'`);

const genFolderPath = path.join(
    __dirname,
    'src',
    'engine',
    'generated'
)

if (!fs.existsSync(genFolderPath)) {
    fs.mkdirSync(genFolderPath)
}

const genPath = path.join(
    __dirname,
    'src',
    'engine',
    'generated',
    'engineVersion.ts'
);

console.log(`  writing to path: '${genPath}'`);

fs.writeFileSync(
    genPath,
    `
// generated code    
// do not change it manually    
export const ENGINE_VERSION = "${pom.project.version}";

`, { encoding: 'utf8' }
)

console.log('  backend version is written');

process.exit(0)