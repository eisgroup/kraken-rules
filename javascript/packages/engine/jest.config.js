module.exports = {
    moduleFileExtensions: ['ts', 'tsx', 'js'],
    transform: {
        '^.+\\.ts$': ['@swc/jest'],
    },
    testMatch: ['**/?(*.)test.ts'],
    verbose: true,
    cacheDirectory: './target/tmp/',
    coverageDirectory: './target/coverage',
    coverageThreshold: {
        global: {
            branches: 93,
            functions: 96,
            lines: 96,
            statements: 96,
        },
    },
    collectCoverageFrom: [
        '**/*[^d\\.].ts',
        '!**/node_modules/**',
        '!**/target/**',
        '!**/__tests__/**',
        '!**/index.ts',
        '!**/**Error**.ts',
        '!**/Reducer.ts', //interface only
        '!**/RulePayloadHandler.ts', //interface only
        '!**/RepositoryClient.ts', //deprecated
        '!**/info/iterators/**', //deprecated
        '!**/config.ts',
        '!**/DevelopmentLogger.ts',
    ],
    testEnvironment: 'node',
    setupFilesAfterEnv: ['./__tests__/console.setup.js', './__tests__/matchers.setup.ts'],
}
