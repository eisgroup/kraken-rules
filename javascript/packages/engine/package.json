{
    "name": "kraken-typescript-engine",
    "version": "1.55.1-stage.0",
    "description": "",
    "main": "target/dist/js/src/index.js",
    "types": "target/dist/definitions/src/index.d.ts",
    "author": "Pavel Surinin <psurinin@eisgroup.com> (http://eisgroup.com)",
    "license": "Apache-2.0",
    "publishConfig": {
        "registry": "https://genesis-npm-release.exigengroup.com/repository/genesis-npm-release/"
    },
    "typedocMain": "src/index.ts",
    "dependencies": {
        "bignumber.js": "9.0.1",
        "declarative-js": "3.8.15",
        "fast-deep-equal": "2.0.1",
        "kraken-engine-api": "1.55.1-stage.0",
        "kraken-model": "1.55.1-stage.0",
        "moment": "2.29.4"
    },
    "devDependencies": {
        "@types/jest": "27.5.1",
        "@types/lodash": "4.14.116",
        "@types/node": "13.9.0",
        "child_process": "1.0.2",
        "fast-xml-parser": "^3.19.0",
        "jest": "27.5.1",
        "jsdom": "11.3.0",
        "jsdom-global": "3.0.2",
        "kraken-jest-matchers": "1.55.1-stage.0",
        "kraken-model-builder": "1.55.1-stage.0",
        "kraken-test-product": "1.55.1-stage.0",
        "kraken-test-product-model-tree": "1.55.1-stage.0",
        "prettier": "1.14.3",
        "rimraf": "3.0.2",
        "ts-node": "4.1.0",
        "typescript": "5.1.6"
    },
    "scripts": {
        "ci-versions": "echo \"typescript version\" && tsc -v && echo \"jest version\" && jest -v",
        "test": "jest -c jest.config.js",
        "test:coverage": "jest -c jest.config.js --coverage",
        "lint": "eslint src __tests__",
        "compile": "node generate-version && tsc --pretty",
        "postcompile": "yarn transpile",
        "transpile": "echo 'no transpile'",
        "del:compiled": "rimraf target",
        "build": "yarn run compile && yarn run lint && yarn run test:coverage",
        "prepare": "yarn run ci-versions && yarn run del:compiled && yarn run compile && yarn run lint -- --fix && yarn run test:coverage",
        "release": "yarn publish"
    },
    "files": [
        "target/dist/definitions",
        "target/dist/js"
    ],
    "gitHead": "efcf1a5e805088ee8af67ef21c2b8ce2fea62eb4"
}