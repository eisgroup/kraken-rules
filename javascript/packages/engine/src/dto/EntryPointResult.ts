/*
 *  Copyright 2018 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { FieldEvaluationResult } from "./FieldEvaluationResult";
import { RuleEvaluationResults } from "./RuleEvaluationResults";
import RuleEvaluationResult = RuleEvaluationResults.RuleEvaluationResult;

import { Reducer } from "declarative-js";
import flat = Reducer.flat;
import { ConditionEvaluationResult } from "./ConditionEvaluationResult";

/**
 * Contains evaluation results from specific entry point
 */
export class EntryPointResult {
    private readonly results: Record<string, FieldEvaluationResult>;
    readonly evaluationTimestamp: Date;
    constructor(results: Record<string, FieldEvaluationResult>) {
        this.results = results;
        this.getApplicableResults.bind(this);
        this.evaluationTimestamp = new Date();
    }
    /**
     * Flatten contents into single list of {@link RuleEvaluationResult}s
     */
    getAllRuleResults(): RuleEvaluationResult[] {
        return Object.values(this.results)
            .map(x => x.ruleResults)
            .reduce(flat, []);
    }
    getFieldResults(): Record<string, FieldEvaluationResult> {
        return this.results;
    }

    getApplicableResults(): RuleEvaluationResult[] {
        return this.getAllRuleResults()
            .filter(r => ConditionEvaluationResult.isApplicable(r.conditionEvaluationResult));
    }

    static empty(): EntryPointResult {
        return new EntryPointResult({});
    }
}
