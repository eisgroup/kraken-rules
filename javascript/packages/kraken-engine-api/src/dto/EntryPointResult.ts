/*
 *  Copyright 2022 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { FieldEvaluationResult } from './FieldEvaluationResult'
import { RuleEvaluationResults } from './RuleEvaluationResults'
import RuleEvaluationResult = RuleEvaluationResults.RuleEvaluationResult

/**
 * Contains evaluation results from specific entry point
 */
export interface EntryPointResult {
    readonly evaluationTimestamp: Date
    readonly ruleTimeZoneId: string

    /**
     * Returns entry point result mapped by field name.
     */
    getFieldResults(): Record<string, FieldEvaluationResult>

    /**
     * Flattens and returns field results into an array of {@link RuleEvaluationResult}
     */
    getAllRuleResults(): RuleEvaluationResult[]

    /**
     * Flattens and returns field results into an array of {@link RuleEvaluationResult}
     * filtered by rule condition evaluation status.
     */
    getApplicableResults(): RuleEvaluationResult[]
}
