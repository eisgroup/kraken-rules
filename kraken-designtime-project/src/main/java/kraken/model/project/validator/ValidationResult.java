/*
 *  Copyright 2017 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package kraken.model.project.validator;

import java.util.Collections;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

import org.slf4j.Logger;

import static kraken.model.project.validator.Severity.ERROR;
import static kraken.model.project.validator.Severity.INFO;
import static kraken.model.project.validator.Severity.WARNING;

/**
 * @author mulevicius
 */
public class ValidationResult {

    private List<ValidationMessage> validationMessages;

    public ValidationResult(List<ValidationMessage> validationMessages) {
        this.validationMessages = Objects.requireNonNull(validationMessages);
    }

    public List<ValidationMessage> getErrors() {
        return validationMessages.stream().filter(m -> m.getSeverity() == ERROR).collect(Collectors.toList());
    }

    public List<ValidationMessage> getWarnings() {
        return validationMessages.stream().filter(m -> m.getSeverity() == WARNING).collect(Collectors.toList());
    }

    public List<ValidationMessage> getInfo() {
        return validationMessages.stream().filter(m -> m.getSeverity() == INFO).collect(Collectors.toList());
    }

    public List<ValidationMessage> getValidationMessages() {
        return Collections.unmodifiableList(validationMessages);
    }

    public void logMessages(Logger logger) {
        validationMessages.stream().forEach(m -> {
            if (m.getSeverity() == INFO) {
                logger.info(m.toString());
            }
            if (m.getSeverity() == WARNING) {
                logger.warn(m.toString());
            }
            if (m.getSeverity() == ERROR) {
                logger.error(m.toString());
            }
        });
    }

    @Override
    public String toString() {
        return validationMessages.stream().map(ValidationMessage::toString).collect(Collectors.joining(System.lineSeparator()));
    }
}
