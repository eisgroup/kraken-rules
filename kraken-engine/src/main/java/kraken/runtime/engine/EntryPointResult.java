package kraken.runtime.engine;

import kraken.annotations.API;
import kraken.runtime.engine.dto.FieldEvaluationResult;
import kraken.runtime.engine.dto.RuleEvaluationResult;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

/**
 * Contains evaluation results from specific entry point
 *
 * @author rimas
 * @since 1.0
 */
@API
public class EntryPointResult {

    /**
     * Field evaluation results, mapped to field id string
     */
    private Map<String, FieldEvaluationResult> fieldResults;

    private transient LocalDateTime evaluationTimeStamp;
    /**
     * Default constructor
     */
    public EntryPointResult() {
        fieldResults = new HashMap<>();
        evaluationTimeStamp = LocalDateTime.now();
    }

    public EntryPointResult(Map<String, FieldEvaluationResult> fieldResults, LocalDateTime evaluationTimeStamp) {
        this.fieldResults = new HashMap<>(fieldResults);
        this.evaluationTimeStamp = evaluationTimeStamp;
    }

    /**
     * Flatten contents into single list of {@link RuleEvaluationResult}s
     *
     * @return  list of all contained {@link RuleEvaluationResult}s
     */
    public List<RuleEvaluationResult> getAllRuleResults() {
        return fieldResults.entrySet().stream()
                .flatMap(e -> e.getValue().getRuleResults().stream())
                .collect(Collectors.toList());
    }

    public List<RuleEvaluationResult> getApplicableRuleResults() {
        return fieldResults.entrySet().stream()
                .flatMap(e -> e.getValue().getRuleResults().stream())
                .filter(r -> r.getConditionEvaluationResult().isApplicable())
                .collect(Collectors.toList());
    }

    public LocalDateTime getEvaluationTimeStamp() {
        return evaluationTimeStamp;
    }

    public Map<String, FieldEvaluationResult> getFieldResults() {
        return fieldResults;
    }

}
