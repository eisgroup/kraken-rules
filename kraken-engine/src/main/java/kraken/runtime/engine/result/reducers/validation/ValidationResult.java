/*
 *  Copyright 2019 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package kraken.runtime.engine.result.reducers.validation;

import java.util.List;

import kraken.annotations.API;
import kraken.model.validation.ValidationSeverity;
import kraken.runtime.engine.dto.ContextFieldInfo;

/**
 * Represents single error message
 *
 * @author rimas
 * @since 1.0
 */
@API
public class ValidationResult {

    /**
     * Name of the rule which generated error message
     */
    private String ruleName;

    /**
     * Error message text.
     * If message is a template then this is already a formatted message
     * with all placeholders replaced with template variables.
     */
    private String message;

    /**
     * Error message code
     */
    private String messageCode;

    /**
     * If message is a template then this is an ordered list of evaluated and formatted template placeholder expressions.
     * Can be used as a values for indexed placeholders when formatting localized message.
     */
    private List<String> templateVariables;

    /**
     * Information about field, instance containing this field
     */
    private ContextFieldInfo contextFieldInfo;

    /**
     * Error message severity
     */
    private ValidationSeverity severity;

    /**
     * @deprecated use {@link #ValidationResult(String, String, String, List, ValidationSeverity, ContextFieldInfo)} instead.
     */
    @Deprecated(since = "1.14.0", forRemoval = true)
    public ValidationResult(String ruleName,
                            String message,
                            String messageCode,
                            ValidationSeverity severity,
                            ContextFieldInfo contextFieldInfo) {
        this(ruleName, message, messageCode, List.of(), severity, contextFieldInfo);
    }

    public ValidationResult(String ruleName,
                            String message,
                            String messageCode,
                            List<String> templateVariables,
                            ValidationSeverity severity,
                            ContextFieldInfo contextFieldInfo) {
        this.ruleName = ruleName;
        this.message = message;
        this.messageCode = messageCode;
        this.templateVariables = templateVariables;
        this.contextFieldInfo = contextFieldInfo;
        this.severity = severity;
    }

    public String getRuleName() {
        return ruleName;
    }

    public String getMessage() {
        return message;
    }

    public String getMessageCode() {
        return messageCode;
    }

    public List<String> getTemplateVariables() {
        return templateVariables;
    }

    public ValidationSeverity getSeverity() {
        return severity;
    }

    public ContextFieldInfo getContextFieldInfo() {
        return contextFieldInfo;
    }

}
