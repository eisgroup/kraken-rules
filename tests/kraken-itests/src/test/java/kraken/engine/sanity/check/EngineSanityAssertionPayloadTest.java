/*
 *  Copyright 2019 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package kraken.engine.sanity.check;

import kraken.runtime.engine.EntryPointResult;
import kraken.runtime.engine.result.reducers.validation.ValidationResult;
import kraken.testproduct.domain.*;
import kraken.utils.MockAutoPolicyBuilder;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Collections;
import java.util.List;

import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;
import static kraken.testing.matchers.KrakenMatchers.*;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.beans.HasPropertyWithValue.hasProperty;
import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
import static org.hamcrest.MatcherAssert.assertThat;
import static kraken.test.KrakenItestMatchers.matchesSnapshot;

/**
 * This test is sanity check test, which results must match ts-engine test results.
 * Tests on both engines should match same asserts.
 *
 * @author psurinin
 * @author avasiliauskas
 * @since 1.0
 */
public final class EngineSanityAssertionPayloadTest extends SanityEngineBaseTest {

    @BeforeClass
    public static void beforeAll() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }

    @Test
    public void shouldExecuteAssertionAutoPolicyEntryPointWithValidData() {
        Policy policy = new MockAutoPolicyBuilder().addValidAutoPolicyWithMockDateTime().build();
        final EntryPointResult result = engine.evaluate(policy, "AssertionAutoPolicy");

        assertThat(result, hasRuleResults(6));
        assertThat(result, hasNoValidationFailures());
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }

    @Test
    public void shouldExecuteAssertionAutoPolicyEntryPointWithEmptyData() {
        Policy policy = new MockAutoPolicyBuilder().addEmptyAutoPolicy().build();
        final EntryPointResult result = engine.evaluate(policy, "AssertionAutoPolicy");

        assertThat(result.getAllRuleResults(), hasSize(4)); 
        assertThat(result, hasValidationFailures(3));
        assertThat(result, ignoredRuleCountIs(1));
    }

    @Test
    public void shouldExecuteAssertionAutoPolicyEntryPointWithNotValidData() {
        final PersonInfo personInfo = new PersonInfo("Antanas", "Antanas", new AddressInfo());
        Policy policy = new MockAutoPolicyBuilder()
                .addTermDetails(new TermDetails(101))
                .addBillingInfo(new BillingInfo())
                .addRiskItems(Collections.singletonList(new Vehicle()))
                .addParty(new Party(personInfo))
                .build();

        final EntryPointResult result = engine.evaluate(policy, "AssertionAutoPolicy");
        final List<ValidationResult> errorResults = validationStatusReducer.reduce(result).getErrorResults();

        assertThat(result.getAllRuleResults(), hasSize(4)); 
        assertThat(errorResults, containsInAnyOrder(
                hasProperty("ruleName", is("R0155")),
                hasProperty("ruleName", is("R0150")),
                hasProperty("ruleName", is("R0051")),
                hasProperty("ruleName", is("R0073"))
        ));
        assertThat(result, hasValidationFailures(4));
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }

    @Test
    public void shouldExecuteAssertionAutoPolicyEntrypointWithNotValidDataWithRestriction() {
        final Vehicle vehicle = new Vehicle();
        vehicle.setId("-1");

        final PersonInfo personInfo = new PersonInfo("Antanas", "Antanas", new AddressInfo());
        Policy policy = new MockAutoPolicyBuilder()
                .addTermDetails(new TermDetails(101))
                .addBillingInfo(new BillingInfo())
                .addRiskItems(Collections.singletonList(vehicle))
                .addParty(new Party(personInfo))
                .build();
        final EntryPointResult result = engine.evaluateSubtree(
                policy,
                policy.getRiskItems().get(0),
                "AssertionAutoPolicy"
        );
        assertThat(result.getAllRuleResults(), hasSize(1));
        assertThat(result.getAllRuleResults().get(0).getRuleInfo().getContext(), is("Vehicle"));

        assertThat(result, hasValidationFailures(1));
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }

    @Test
    public void shouldEvaluateAssertionAutoPolicyWithNodeThatHasNoRule() {
        Policy policy = new MockAutoPolicyBuilder().addValidAutoPolicyWithMockDateTime().build();
        Insured insured = new Insured();
        policy.setInsured(insured);
        final EntryPointResult result = engine.evaluateSubtree(policy, insured, "AssertionAutoPolicy");

        assertThat(result.getAllRuleResults(), hasSize(0)); 
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }
}
