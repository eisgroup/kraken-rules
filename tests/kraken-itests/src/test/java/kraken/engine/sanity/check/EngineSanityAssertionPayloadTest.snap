kraken.engine.sanity.check.EngineSanityAssertionPayloadTest.shouldEvaluateAssertionAutoPolicyWithNodeThatHasNoRule=[
  {}
]


kraken.engine.sanity.check.EngineSanityAssertionPayloadTest.shouldExecuteAssertionAutoPolicyEntryPointWithNotValidData=[
  {
    "AddressInfo:-1:doNotSolicit": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "doNotSolicit",
        "fieldPath": "doNotSolicit"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": false,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "PersonInfo:-1:firstName": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "PersonInfo",
        "fieldName": "firstName",
        "fieldPath": "firstName"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0150",
            "context": "PersonInfo",
            "targetPath": "firstName",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": false,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: firstName !\u003d lastName",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: firstName !\u003d lastName",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:termNo": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "termNo",
        "fieldPath": "termDetails.termNo"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0051",
            "context": "Policy",
            "targetPath": "termNo",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": false,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: termNo \u003e\u003d 0 and termNo \u003c 100",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: termNo \u003e\u003d 0 and termNo \u003c 100",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Vehicle:-1:included": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Vehicle",
        "fieldName": "included",
        "fieldPath": "included"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0073",
            "context": "Vehicle",
            "targetPath": "included",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": false,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: included \u003d true or included \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: included \u003d true or included \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    }
  }
]


kraken.engine.sanity.check.EngineSanityAssertionPayloadTest.shouldExecuteAssertionAutoPolicyEntryPointWithValidData=[
  {
    "AddressInfo:-1:doNotSolicit": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "doNotSolicit",
        "fieldPath": "doNotSolicit"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "AddressInfo:2:doNotSolicit": {
      "contextFieldInfo": {
        "contextId": "2",
        "contextName": "AddressInfo",
        "fieldName": "doNotSolicit",
        "fieldPath": "doNotSolicit"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "BillingAddress:-1:doNotSolicit": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "BillingAddress",
        "fieldName": "doNotSolicit",
        "fieldPath": "doNotSolicit"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "PersonInfo:-1:firstName": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "PersonInfo",
        "fieldName": "firstName",
        "fieldPath": "firstName"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0150",
            "context": "PersonInfo",
            "targetPath": "firstName",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: firstName !\u003d lastName",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: firstName !\u003d lastName",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Policy:-1:termNo": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "termNo",
        "fieldPath": "termDetails.termNo"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0051",
            "context": "Policy",
            "targetPath": "termNo",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: termNo \u003e\u003d 0 and termNo \u003c 100",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: termNo \u003e\u003d 0 and termNo \u003c 100",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    },
    "Vehicle:-1:included": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Vehicle",
        "fieldName": "included",
        "fieldPath": "included"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0073",
            "context": "Vehicle",
            "targetPath": "included",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: included \u003d true or included \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: included \u003d true or included \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    }
  }
]


kraken.engine.sanity.check.EngineSanityAssertionPayloadTest.shouldExecuteAssertionAutoPolicyEntrypointWithNotValidDataWithRestriction=[
  {
    "Vehicle:-1:included": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Vehicle",
        "fieldName": "included",
        "fieldPath": "included"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0073",
            "context": "Vehicle",
            "targetPath": "included",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": false,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: included \u003d true or included \u003d false",
            "templateVariables": [],
            "messageTemplate": "Value didn\u0027\u0027t match asserton: included \u003d true or included \u003d false",
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          },
          "ruleEvaluationStatus": "APPLIED"
        }
      ]
    }
  }
]